### PR TITLE
Clean up Old RPC Endpoints

### PR DESCRIPTION
**Pull request by terencechain**
This PR cleans up the old and unused RPC end points. We should not be maintaining them as it's no longer necessary

### CODE REVIEW

File: beacon-chain/rpc/beacon_server.go [hunk 1]
```diff
@@ -86,127 +84,6 @@ func (bs *BeaconServer) CanonicalHead(ctx context.Context, req *ptypes.Empty) (*
 	return block, nil
 }
 
-// LatestAttestation streams the latest processed attestations to the rpc clients.
-func (bs *BeaconServer) LatestAttestation(req *ptypes.Empty, stream pb.BeaconService_LatestAttestationServer) error {
```
Comment (terencechain) [inline beacon-chain/rpc/beacon_server.go]: no longer used

File: beacon-chain/rpc/beacon_server.go [hunk 2]
```diff
@@ -86,127 +84,6 @@ func (bs *BeaconServer) CanonicalHead(ctx context.Context, req *ptypes.Empty) (*
...
-			return nil
-		case <-bs.ctx.Done():
-			log.Debug("RPC context closed, exiting goroutine")
-			return nil
-		}
-	}
-}
-
-// DomainData fetches the current domain version information from the beacon state.
-func (bs *BeaconServer) DomainData(ctx context.Context, request *pb.DomainRequest) (*pb.DomainResponse, error) {
```
Comment (terencechain) [inline beacon-chain/rpc/beacon_server.go]: moved to `ValidatorServer`

File: beacon-chain/rpc/beacon_server.go [hunk 3]
```diff
@@ -86,127 +84,6 @@ func (bs *BeaconServer) CanonicalHead(ctx context.Context, req *ptypes.Empty) (*
...
-}
-
-// Eth1Data is a mechanism used by block proposers vote on a recent Ethereum 1.0 block hash and an
-// associated deposit root found in the Ethereum 1.0 deposit contract. When consensus is formed,
-// state.latest_eth1_data is updated, and validator deposits up to this root can be processed.
-// The deposit root can be calculated by calling the get_deposit_root() function of
-// the deposit contract using the post-state of the block hash.
-//
-// TODO(#2307): Refactor for v0.6.
-func (bs *BeaconServer) Eth1Data(ctx context.Context, _ *ptypes.Empty) (*pb.Eth1DataResponse, error) {
```
Comment (terencechain) [inline beacon-chain/rpc/beacon_server.go]: moved to `ProposerServer`

File: beacon-chain/rpc/beacon_server.go [hunk 4]
```diff
@@ -86,127 +84,6 @@ func (bs *BeaconServer) CanonicalHead(ctx context.Context, req *ptypes.Empty) (*
...
-	return nil, nil
-}
-
-// PendingDeposits returns a list of pending deposits that are ready for
-// inclusion in the next beacon block.
-func (bs *BeaconServer) PendingDeposits(ctx context.Context, _ *ptypes.Empty) (*pb.PendingDepositsResponse, error) {
```
Comment (terencechain) [inline beacon-chain/rpc/beacon_server.go]: moved to `ProposerServer`

File: beacon-chain/rpc/proposer_server.go [hunk 5]
```diff
@@ -28,25 +32,6 @@ type ProposerServer struct {
 	canonicalStateChan chan *pbp2p.BeaconState
 }
 
-// ProposerIndex sends a response to the client which returns the proposer index for a given slot. Validators
-// are shuffled and assigned slots to attest/propose to. This method will look for the validator that is assigned
-// to propose a beacon block at the given slot.
-func (ps *ProposerServer) ProposerIndex(ctx context.Context, req *pb.ProposerIndexRequest) (*pb.ProposerIndexResponse, error) {
```
Comment (terencechain) [inline beacon-chain/rpc/proposer_server.go]: no longer needed

File: validator/client/validator_propose.go [hunk 6]
```diff
@@ -21,41 +21,42 @@ import (
 // chain node to construct the new block. The new block is then processed with
 // the state root computation, and finally signed by the validator before being
 // sent back to the beacon node for broadcasting.
-func (v *validator) ProposeBlock(ctx context.Context, slot uint64, idx string) {
+func (v *validator) ProposeBlock(ctx context.Context, slot uint64, pk string) {
```
Comment (terencechain) [inline validator/client/validator_propose.go]: doesn't make sense to call it `idx`

**Comment by terencechain** [comment]
> Can we hold off on reorganizing API service methods?
> We are working on a larger refactoring next week so all of this is likely to change.

Sure. Can we start within the scope of these two actions? (1. proposer propose 2. attester attest)
Reason being we urgently need to update validator run time to 0.6 and not sure if it should be blocked by the larger refactoring effort...

**Comment by prestonvanloon** [comment]
Yes. If you need to make refactoring changes now to unblock then go ahead with that.

File: validator/client/validator_propose.go [hunk 7]
```diff
@@ -21,41 +21,42 @@ import (
...
 	}
 	ctx, span := trace.StartSpan(ctx, "validator.ProposeBlock")
 	defer span.End()
-	span.AddAttributes(trace.StringAttribute("validator", fmt.Sprintf("%#x", v.keys[idx].PublicKey.Marshal())))
-	truncatedPk := idx
-	if len(idx) > 12 {
-		truncatedPk = idx[:12]
+	span.AddAttributes(trace.StringAttribute("validator", fmt.Sprintf("%#x", v.keys[pk].PublicKey.Marshal())))
+	truncatedPk := pk
+	if len(pk) > 12 {
```
Comment (nisdas) [inline validator/client/validator_propose.go]: I think we have a helper that does this, we should use that here too

File: beacon-chain/rpc/beacon_server.go (hunk 1 above)
Comment (nisdas) [reply beacon-chain/rpc/beacon_server.go]: what was the initial reason for using this ?

Comment (terencechain) [reply beacon-chain/rpc/beacon_server.go]: No idea, but its functionality is almost the same as `PendingAttestations`. We really don't need this 😅

File: validator/client/validator_propose.go (hunk 7 above)
Comment (terencechain) [reply validator/client/validator_propose.go]: Ah. Yes. Done

**Comment by terencechain** [comment]
@nisdas ready again :)

**Pull request merged**